File: Java/src/main/java/br/com/locaweb/locamail/api/model/Email.java
package br.com.locaweb.locamail.api.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "T_LCW_EMAIL")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class Email {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_email")
    private Long id_email;

    @Column(name = "id_usuario", nullable = false)
    private Long id_usuario;

    @Column(name = "remetente", nullable = false)
    private String remetente;

    @Column(name = "destinatario", nullable = false)
    private String destinatario;

    @Column(name = "cc", nullable = false)
    private String cc;

    @Column(name = "cco", nullable = false)
    private String cco;

    @Column(name = "assunto", nullable = false)
    private String assunto;

    @Column(name = "corpo", nullable = false)
    private String corpo;

    @Column(name = "editavel", nullable = false)
    private Boolean editavel;

    @Column(name = "enviado", nullable = false)
    private Boolean enviado;

    @Column(name = "horario", nullable = false)
    private String horario;

    @Column(name = "data", nullable = false)
    private String data;

    @Column(name = "agenda_atrelada", nullable = false)
    private Boolean agenda_atrelada;
}

File: Kotlin/app/src/main/java/br/com/fiap/locawebmailapp/model/Email.kt
package br.com.fiap.locawebmailapp.model

import androidx.compose.runtime.Immutable
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import java.time.LocalDate
import java.time.LocalTime
import java.time.format.DateTimeFormatter

@Immutable
@Entity(tableName = "T_LCW_EMAIL",
    foreignKeys =
    [ForeignKey(
        entity = Usuario::class,
        childColumns = ["id_usuario"],
        parentColumns = ["id_usuario"]
    )
    ])
data class Email(
    @PrimaryKey(autoGenerate = true) var id_email: Long = 0,
    var id_usuario: Long = 0,
    var remetente: String = "",
    var destinatario: String = "",
    var cc: String = "",
    var cco: String = "",
    var assunto: String = "",
    var corpo: String = "",
    var editavel: Boolean = false,
    var enviado: Boolean = false,
    var horario: String = LocalTime.now().format(DateTimeFormatter.ofPattern("HH:mm")),
    var data: String = "${LocalDate.now()}",
    var agenda_atrelada: Boolean = false,
    var is_spam: Boolean = false
)